### PR TITLE
auto completion can be toggled on and off

### LINKED ISSUE #60: Disabling auto completion suggestions
As a text writer that uses the tool to write text that does not consist of Smalltalk code, I want an option to disable the Squeak auto completion of words, in order to increase my productivity with fewer distractions. 

**Acceptance criteria:**
- [x] In the preference browser, a category for SquidWord exists
- [x] It contains the option to disable autocompletion in SquidWord windows
- [x] Whenever the option is enabled, no autocompletion window when something is typed in SquidWord.

### PR DESCRIPTION
**Pull request by frcroth**
Should fix #60

### CODE REVIEW

**Comment by sirkrypt0** [comment]
I assume there is no way to test this?

**Review by sirkrypt0 — APPROVED** [review]

**Comment by frcroth** [comment]
> I assume there is no way to test this?

@sirkrypt0 Maybe you could try to type and then check number of morphs / number of text suggestion morphs while typing?

**Comment by LeonMatthes** [comment]
If you want to test the Autocompletion, make sure it's installed first.
You can add it as a dependency of your test package in your Baseline (repo is [here](https://github.com/mrmodder/autocompletion) ).

The API of the Autocompletion is mostly not guaranteed, but it still doesn't change very often.

You can access the Autocompletions controller using `Model>>#completionController` and then use `ECController>>#isMenuOpen` to check whether the autocompletion is opened.

**Review by konradh — COMMENTED** [review]
I'm not really sure why this is necessary as `RichTextEditorModel>>#initializeCompletionController` is now exactly the same as `Model>>#initializeCompletionController`, which we inherit anyway.
It should make absolutely no difference.
Enabling and disabling autocompletion already worked for me before.

**Comment by frcroth** [comment]
> 
> 
> I'm not really sure why this is necessary as `RichTextEditorModel>>#initializeCompletionController` is now exactly the same as `Model>>#initializeCompletionController`, which we inherit anyway.
> It should make absolutely no difference.
> Enabling and disabling autocompletion already worked for me before.

It doesn't seem to work for anyone else usually, but on this branch it does work for me

**Comment by sirkrypt0** [comment]
> If you want to test the Autocompletion, make sure it's installed first.
> You can add it as a dependency of your test package in your Baseline (repo is [here](https://github.com/mrmodder/autocompletion) ).
> 
> The API of the Autocompletion is mostly not guaranteed, but it still doesn't change very often.
> 
> You can access the Autocompletions controller using `Model>>#completionController` and then use `ECController>>#isMenuOpen` to check whether the autocompletion is opened.

I think we should do that to really make sure it can be toggled off and on reliably.

**Comment by sirkrypt0** [comment]
> > I'm not really sure why this is necessary as `RichTextEditorModel>>#initializeCompletionController` is now exactly the same as `Model>>#initializeCompletionController`, which we inherit anyway.
> > It should make absolutely no difference.
> > Enabling and disabling autocompletion already worked for me before.
> 
> It doesn't seem to work for anyone else usually, but on this branch it does work for me

I agree, it works here

File: packages/RichTextEditing-Tests.package/RichTextEditorTests.class/instance/enableAutocompletion.st [hunk 1]
```diff
@@ -0,0 +1,5 @@
+utilities
+enableAutocompletion
+
+	autocompletionPreference := RichTextEditorModel autocompletion.
```
Comment (j-hellenberg) [inline packages/RichTextEditing-Tests.package/RichTextEditorTests.class/instance/enableAutocompletion.st:4]: Why no accessors?

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/initializeCompletionController.st [hunk 2]
```diff
@@ -0,0 +1,6 @@
+initialize-release
+initializeCompletionController
+
+	^ self wantsAutocompletion
+		ifTrue: [self addDependent: self createCompletionController]
+		ifFalse: [nil]
```
Comment (j-hellenberg) [inline packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/initializeCompletionController.st]: This is always called when a new SquidWord is opened?

Comment (frcroth) [reply packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/initializeCompletionController.st]: Yes should be

File: packages/RichTextEditing-Tests.package/RichTextEditorTests.class/instance/enableAutocompletion.st (hunk 1 above)
Comment (frcroth) [reply packages/RichTextEditing-Tests.package/RichTextEditorTests.class/instance/enableAutocompletion.st:4]: not necessary

Comment (j-hellenberg) [reply packages/RichTextEditing-Tests.package/RichTextEditorTests.class/instance/enableAutocompletion.st:4]: I thought we did settle on always using them

File: packages/RichTextEditing-Tests.package/RichTextEditorTests.class/instance/enableAutocompletion.st [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+	RichTextEditorModel autocompletion: true.
```
Comment (sirkrypt0) [inline packages/RichTextEditing-Tests.package/RichTextEditorTests.class/instance/enableAutocompletion.st]: Remove dot
```suggestion
	RichTextEditorModel autocompletion: true
```

**Review by sirkrypt0 — CHANGES_REQUESTED** [review]

**Review by sirkrypt0 — APPROVED** [review]

**Comment by frcroth** [comment]
todo: Test should really test for autocompletion (acceptance test)

**Comment by sirkrypt0** [comment]
maybe consider adding these changes on a new branch to avoid merge conflicts

**Pull request merged**